File: 0735_Asteroid Collision/AsteroidCollision.py
class Solution(object):
    def asteroidCollision(self, asteroids):
        """
        :type asteroids: List[int]
        :rtype: List[int]
        """
        stack = []
        for asteroid in asteroids:
            if asteroid > 0 or len(stack) == 0:
                stack.append(asteroid)
                continue
            while True:
                if stack[len(stack)-1] < 0:
                    stack.append(asteroid)
                    break

                if stack[len(stack)-1] > -asteroid:
                    break
                    
                if stack[len(stack)-1] == -asteroid:
                    stack.pop()
                    break          
                    
                stack.pop()
                
                if len(stack) == 0:
                    stack.append(asteroid)
                    break

        return stack

File: 0735_Asteroid Collision/AsteroidCollision.cs
    public class Solution
    {
        public int[] AsteroidCollision(int[] asteroids)
        {
            var stack = new Stack<int>();
            foreach (var asteroid in asteroids)
            {
                if(stack.Count == 0 || asteroid > 0)
                {
                    stack.Push(asteroid);
                    continue;
                }
 
                while (true)
                {
                    if(stack.Peek() < 0)
                    {
                        stack.Push(asteroid);
                        break;
                    }
 
                    if(stack.Peek() == -asteroid)
                    {
                        stack.Pop();
                        break;
                    }
 
                    if(stack.Peek() > -asteroid)
                    {
                        break;
                    }
 
                    stack.Pop();
 
                    if(stack.Count == 0)
                    {
                        stack.Push(asteroid);
                        break;
                    }
                }
 
            }
 
            return stack.ToArray().Reverse().ToArray();
        }
    }
}
